### PR TITLE
Bug fixes for issue 54, 53 and 51.

### PR DESCRIPTION
**Pull request by cbommas**
Added test for issue 51.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Change Log - AWS AppSync SDK for Android
 
+## [Release 2.6.27](https://github.com/awslabs/aws-mobile-appsync-sdk-android/releases/tag/release_v2.6.27)
+
+### Bug Fixes
+* Fixed bug in subscribe call that was setting the QOS to be 0. It is now correctly set to QOS 1. See [issue #54](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/54)
```
Comment (minbi) [inline CHANGELOG.md]:
```suggestion
* Fixed bug in subscribe call that was setting the QoS to be 0. It is now correctly set to QoS 1. See [issue #54](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/54)
```

Comment (minbi) [reply CHANGELOG.md]: I've seen it stylized as QoS

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+* Fixed bug that was causing a ConcurrentModificationException in the logic to handle connection loss. See [PR #22](https://github.com/awslabs/aws-mobile-appsync-sdk-android/pull/22) and [issue #53](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/53)
```
Comment (minbi) [inline CHANGELOG.md]:
```suggestion
* Fixed bug that was causing a ConcurrentModificationException in the logic to handle connection loss. See [issue #22](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/22) and [issue #53](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/53)
```

File: CHANGELOG.md (hunk 1 above)
Comment (cbommas) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (cbommas) [reply CHANGELOG.md]: Done. It was PR 41 for issue 22 - I have changed it to the right PR#.

**Pull request merged**